src/irmin/type/type_size.ml

```diff
@@ -46,13 +46,25 @@ let bool (_ : bool) = 1
 
 let float (_ : float) = 8 (* NOTE: we consider 'double' here *)
 
-let string ?(headers = true) n s =
+let boxed_string n s =
   let s = String.length s in
-  if not headers then s else len s n + s
+  len s n + s
 
-let bytes ?(headers = true) n s =
+let unboxed_string = function
+  | `Fixed len -> fun _ -> len (* fixed-size strings are never boxed *)
+  | _ -> String.length
```

Comment:
Not related to this PR, but I think it's unintuitive that ``Type.(string_of (`Fixed 8))` and `Type.(string_of `Int64)`` are not the same w.r.t. this function (since the corresponding string doesn't need to be of the correct length). I think we should make a consistent choice here, but we could also just say that it's undefined.